src/dune_pkg/opam.mli

```diff
@@ -0,0 +1,39 @@
+open Stdune
+
+module Repo : sig
+  (** An opam repository *)
+  type t
+
+  (** Create a [t] from a path to a local directory containing an opam
+      repository. This should be a directory with a subdirectory named
+      "packages" with one subdirectory for each package name, each with
+      subdirectories for each version. *)
```

Comment:
I think stating "this should be a valid opam repository" saves us from defining what an opam repository is and potentially creating our own version of opam-repositories that don't work with OPAM.

Reply:
Good point. I'll update the comment and help message. This is also a good reason to enforce that a "repo" file is present inside the opam repository even if it doesn't contribute to solving.